runtime/vm/UpcallVMHelpers.cpp

```diff
@@ -140,7 +140,7 @@ native2InterpJavaUpcallD(J9UpcallMetaData *data, void *argsListPointer)
 U_8 * JNICALL
 native2InterpJavaUpcallStruct(J9UpcallMetaData *data, void *argsListPointer)
 {
-	UDATA returnValue = native2InterpJavaUpcallImpl(data, argsListPointer);
+	U_64 returnValue = native2InterpJavaUpcallImpl(data, argsListPointer);
 	return (U_8 *)returnValue;
```

Comment:
This should cast through `UDATA` so we aren't casting to a pointer of a different size:
```
	return (U_8 *)(UDATA)returnValue;
```

Reply:
Fixed.